packages/@react-aria/table/src/useTableColumnResize.ts

```diff
@@ -28,37 +29,78 @@ export interface TableColumnResizeAria {
+  /** Aria label for the hidden input. Gets read when resizing. */
   label: string,
-  triggerRef: RefObject<HTMLDivElement>,
+  /**
+   * Ref to the trigger if resizing was started from a column header menu. If it's provided,
+   * focus will be returned there when resizing is done.
+   * */
+  triggerRef?: RefObject<HTMLDivElement>,
+  /** If resizing is disabled. */
   isDisabled?: boolean,
-  onMove: (e: MoveMoveEvent) => void,
-  onMoveEnd: (e: MoveEndEvent) => void
+  /** If the resizer was moved. Different from onResize because it is always called. */
+  onMove?: (e: MoveMoveEvent) => void,
+  /**
+   * If the resizer was moved. Different from onResizeEnd because it is always called.
+   * It also carries the interaction details in the object.
+   * */
+  onMoveEnd?: (e: MoveEndEvent) => void,
+  /** Called when resizing starts. */
+  onResizeStart: (key: Key) => void,
+  /** Called for every resize event that results in new column sizes. */
+  onResize: (widths: Map<Key, number | string>) => void,
+  /** Called when resizing ends. */
+  onResizeEnd: (key: Key) => void
 }
 
-export function useTableColumnResize<T>(props: AriaTableColumnResizeProps<T>, state: TableState<T>, columnState: TableColumnResizeState<T>, ref: RefObject<HTMLInputElement>): TableColumnResizeAria {
-  let {column: item, triggerRef, isDisabled} = props;
-  const stateRef = useRef<TableColumnResizeState<T>>(null);
-  stateRef.current = columnState;
+export function useTableColumnResize<T>(props: AriaTableColumnResizeProps<T>, state: TableState<T>, layoutState: TableLayoutState, ref: RefObject<HTMLInputElement>): TableColumnResizeAria {
+  let {column: item, triggerRef, isDisabled, onResizeStart, onResize, onResizeEnd} = props;
   const stringFormatter = useLocalizedStringFormatter(intlMessages);
   let id = useId();
+  let isResizing = useRef(false);
+  let lastSize = useRef(null);
 
   let {direction} = useLocale();
   let {keyboardProps} = useKeyboard({
     onKeyDown: (e) => {
-      if (e.key === 'Escape' || e.key === 'Enter' || e.key === ' ' || e.key === 'Tab') {
+      if (triggerRef?.current && (e.key === 'Escape' || e.key === 'Enter' || e.key === ' ' || e.key === 'Tab')) {
         e.preventDefault();
         // switch focus back to the column header on anything that ends edit mode
         focusSafely(triggerRef.current);
       }
     }
   });
 
+  let startResize = useCallback((item) => {
+    if (!isResizing.current) {
+      layoutState.onColumnResizeStart(item.key);
+      onResizeStart?.(item.key);
+    }
+    isResizing.current = true;
+  }, [isResizing, onResizeStart, layoutState]);
+
+  let resize = useCallback((item, newWidth) => {
+    let sizes = layoutState.onColumnResize(item.key, newWidth);
+    onResize?.(sizes);
+    lastSize.current = sizes;
+  }, [onResize, layoutState]);
+
+  let endResize = useCallback((item) => {
+    if (isResizing.current) {
+      layoutState.onColumnResizeEnd(item.key);
+      onResizeEnd?.(lastSize.current);
```

Comment:
Noticed that onResizeEnd is called with `null` if the user just clicks on the resizer without moving it.

Reply:
addressed